manifests/grub.pp

```diff
@@ -0,0 +1,44 @@
+# === Copyright
+#
+# Copyright 2018, Kumina B.V., Tim Stoop
+# Licensed under the Apache License, Version 2.0 (the "License");
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+
+# == Class: os_hardening::grub
+#
+# Hardens the grub config
+#
+class os_hardening::grub (
+  String  $user                  = 'root',
+  String  $password_hash         = false,
+  Boolean $boot_without_password = true,
+) {
+
+  file { '/etc/grub.d/01_hardening':
+    content => template('os_hardening/grub_hardening.erb'),
+    notify  => Exec['Grub configuration recreate for os_hardening::grub'],
+  }
+
+  if $boot_without_password {
```

Comment:
What if you switch from `boot_without_password = true` to `boot_without_password = false`? --> undo the change in the file